pennylane/tape/tapes/tape.py

```diff
@@ -1040,16 +1054,11 @@ def execute_device(self, params, device):
             params (list[Any]): The quantum tape operation parameters. If not provided,
                 the current tape parameter values are used (via :meth:`~.get_parameters`).
         """
-        obs_wires = [wire for m in self.measurements for wire in m.wires if m.obs is not None]
-
-        if len(obs_wires) != len(set(obs_wires)):
-
-            obs = [m.obs for m in self.measurements if m.obs is not None]
-            if not all(len(o.diagonalizing_gates()) == 0 for o in obs):
-                raise qml.QuantumFunctionError(
-                    "Multiple observables are being evaluated on the same wire. Call "
-                    "tape.expand(expand_measurements=True) to support this."
-                )
+        if not all(len(o.diagonalizing_gates()) == 0 for o in self._repeated_observables):
+            raise qml.QuantumFunctionError(
+                "Multiple observables are being evaluated on the same wire. Call tape.expand() "
+                "prior to execution to support this."
+            )
```

Comment:
Long term, another thing to devolve to the devices. especially for simulator devices, where non-communiting observables are no issue 😄